pf4j/src/main/java/org/pf4j/PluginClassLoader.java

```diff
@@ -251,38 +224,37 @@ public URL getResource(String name) {
         return null;
     }
 
-    private URL findResourceFromDependencies(String name) {
+    protected URL findResourceFromDependencies(String name) {
         log.trace("Search in dependencies for resource '{}'", name);
         List<PluginDependency> dependencies = pluginDescriptor.getDependencies();
         for (PluginDependency dependency : dependencies) {
             PluginClassLoader classLoader = (PluginClassLoader) pluginManager.getPluginClassLoader(dependency.getPluginId());
 
             // If the dependency is marked as optional, its class loader might not be available.
-            if (classLoader == null && dependency.isOptional()) {
+            if (classLoader == null || dependency.isOptional()) {
                 continue;
             }
 
             URL url = classLoader.findResource(name);
             if (Objects.nonNull(url)) {
-            	return url;
+                return url;
             }
         }
 
         return null;
     }
 
-    private Collection<URL> findResourcesFromDependencies(String name) throws IOException {
+    protected Collection<URL> findResourcesFromDependencies(String name) throws IOException {
         log.trace("Search in dependencies for resources '{}'", name);
         List<URL> results = new ArrayList<>();
         List<PluginDependency> dependencies = pluginDescriptor.getDependencies();
         for (PluginDependency dependency : dependencies) {
             PluginClassLoader classLoader = (PluginClassLoader) pluginManager.getPluginClassLoader(dependency.getPluginId());
 
             // If the dependency is marked as optional, its class loader might not be available.
-            if (classLoader == null && dependency.isOptional()) {
+            if (classLoader == null || dependency.isOptional()) {
```

Comment:
I think that is not good. You revert the bug solved by #386.

Reply:
oops , solved